stripe/src/main/java/com/stripe/android/model/CardBrand.kt

```diff
@@ -241,6 +252,10 @@ enum class CardBrand(
         return groups
     }
 
+    private fun getPatternForLength(cardNumber: String): Pattern? {
+        return partialPatterns[cardNumber.length] ?: pattern
```

Comment:
`pattern` isn't tight enough if it's the fallback, e.g. If you try a non-JCB card like `351700000...` longer than the partial patterns, I believe the JCB logo still shows up? I'm eyeballing it statically so not sure.

The experience I'm thinking of is the JCB logo should disappear as a hint if you move the cursor back and introduce a typo.